### PR TITLE
LHN list re-render

### PR DESCRIPTION
**Pull request by perunt**
here I'm trying to address the bug described: https://github.com/Expensify/App/pull/36420#issuecomment-1946291081

discussion: https://expensify.slack.com/archives/C05LX9D6E07/p1708011570713419
### Fixed Issues

$ https://github.com/Expensify/App/pull/36420#issuecomment-1946291081
PROPOSAL:

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

NOTE: This issue occurs only once when you navigate to this screen for the first time.

1. Log out of the app.
2. Log back in.
3. Open the settings screen from the Left-Hand Navigation (LHN).
4. Check if there is no noticeable delay in the rendering of the settings screen. 

Alternate method to reproduce the issue:

1. Open the chat tab on the LHN.
2. Restart the Metro bundler.
3. Switch to the settings screen.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/48593211/67827a1b-4aa3-4d49-8892-5487b9e247d8

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/48593211/9427c3c1-ac43-4d9a-92fb-536b927c4db2

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/48593211/904fe7a2-d04d-40f7-b715-a74adb5009b0

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/48593211/b97a5b5b-45a4-40c9-8f53-0dca1ec75bd1

</details>

### CODE REVIEW

File: src/pages/home/sidebar/SidebarScreen/index.js [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
 import FreezeWrapper from '@libs/Navigation/FreezeWrapper';
 import BaseSidebarScreen from './BaseSidebarScreen';
 import sidebarPropTypes from './sidebarPropTypes';
 
 function SidebarScreen(props) {
-    const {isSmallScreenWidth} = useWindowDimensions();
-
     return (
-        <FreezeWrapper keepVisible={!isSmallScreenWidth}>
+        <FreezeWrapper>
```
Comment (WoLewicki) [inline src/pages/home/sidebar/SidebarScreen/index.js:8]: Why did you remove this? Is the sidebar still clickable if you e.g. navigate to different reports twice? I think it was handling a case where you have the sidebar visible all the time and navigate deeper in the main stack, but maybe the navigation structure changed in a way that it is not necessary right now?

File: src/libs/Navigation/FreezeWrapper.tsx [hunk 2]
```diff
@@ -28,8 +27,8 @@ function FreezeWrapper({keepVisible = false, children}: FreezeWrapperProps) {
             // if the screen is more than 1 screen away from the current screen, freeze it,
             // we don't want to freeze the screen if it's the previous screen because the freeze placeholder
             // would be visible at the beginning of the back animation then
-            if (navigation.getState().index - (screenIndexRef.current ?? 0) > 1) {
-                InteractionManager.runAfterInteractions(() => setIsScreenBlurred(true));
+            if (navigation.getState().index - (screenIndexRef.current ?? 0) >= 1) {
```
Comment (WoLewicki) [inline src/libs/Navigation/FreezeWrapper.tsx:30]: Does the issue with beginning of animation not occur anymore? And did you try to run it in the new arch? I think the case with `InteractionManager` was needed there only and I think we will want to enable it on web ASAP too.

**Review by WoLewicki — COMMENTED** [review]
I left some comments, please answer them :tada:

File: src/pages/home/sidebar/SidebarScreen/index.js (hunk 1 above)
Comment (jbroma) [reply src/pages/home/sidebar/SidebarScreen/index.js:8]: FYI I tested this fix locally and the list is still fully clickable on the web, and causes no issues on mobile. It seems plausible that the nav structure changed and this is no loner a case

File: src/libs/Navigation/FreezeWrapper.tsx (hunk 2 above)
Comment (jbroma) [reply src/libs/Navigation/FreezeWrapper.tsx:30]: I can pinpoint the any issues with the animations. I haven't tested with new arch, is this even something that is enabled in the app, because it doesn't seem like it.

Comment (perunt) [reply src/libs/Navigation/FreezeWrapper.tsx:30]: Animation of goBack works ok for me. I haven't tried it with a new arch yet.

**Comment by hayata-suenaga** [comment]
@perunt, is the purpose of this PR to ensure that the hidden screen freezes when there are two or more screens in the stack navigator?

**Comment by hayata-suenaga** [comment]
okay thank you for the confirmation

@situchan please review the PR when you have time 🙇

**Comment by hurali97** [comment]
Hey @perunt 👋 

Great work ! I was testing out your PR alongside with this [PR](https://github.com/Expensify/App/pull/36420) and It works great. Only thing I want to flag is these two PR appears to go together. For example, only with your PR we still see the slowness between switch but there's no such hang observed. Now, to improve the slowness between switch, the linked PR is important.

I just wanted to flag the above and also below is how these two PRs look together.

<details>
<summary>Before</summary>

https://github.com/Expensify/App/assets/47336142/583756a6-8163-4357-9828-e25545c35ef7

</details>

<details>
<summary>After</summary>

https://github.com/Expensify/App/assets/47336142/f61782ce-19cb-4017-b0ff-09d86db7bba2

</details>

**Comment by perunt** [comment]
hey @hurali97
Just to make sure we're on the same page - are you referring to the initial switching? Could you also compare it with subsequent switchings as well?

If not, have you had a chance to test the switching on the main branch, where the issue with multiple renderItem calls doesn't seem to reproduce? Is the performance still an issue in that scenario?

The key concern here is that on the first switch, we encounter a page that completely freezes for 15-20 seconds, which we need to avoid. If there's an alternative solution, I'd be eager to explore it. The difference might be due to `runAfterInteractions`. I'll double-check to see if I can manage to execute it in the next frame without disrupting the freeze logic.

**Comment by hurali97** [comment]
> Just to make sure we're on the same page - are you referring to the initial switching? Could you also compare it with subsequent switchings as well?

I am referring to both the initial switch and the subsequent switches. If you notice in the After video, both the initial and subsequent switches are quick which happens because of the PR I linked. 

What I am trying to say is your PR for sure fixes the multiple renderItem calls but we still have some slowness in screen rendering after the switch. The switch happens quickly but not the rendering. The PR I linked fixes that and now with both of these PRs we have quick switching, no extra renderItem calls and quick rendering as well.


> If not, have you had a chance to test the switching on the main branch, where the issue with multiple renderItem calls doesn't seem to reproduce? Is the performance still an issue in that scenario?

Below you can see how main behaves with your PR changes only. We have no extra renderItem calls and quick switching but delayed rendering. 

<details>
<summary>Main</summary>

https://github.com/Expensify/App/assets/47336142/0fe0ed9a-1c8c-4a69-ae72-5061c7369b2d

</details>


<details>
<summary>Both PRs combined</summary>

https://github.com/Expensify/App/assets/47336142/f61782ce-19cb-4017-b0ff-09d86db7bba2

</details>

**Comment by situchan** [comment]
> What I am trying to say is your PR for sure fixes the multiple renderItem calls but we still have some slowness in screen rendering after the switch. 

@hurali97 at least this PR won't cause another performance regression right?

> we still have some slowness in screen rendering after the switch

this is already existing bug on main right?

**Comment by hurali97** [comment]
> @hurali97 at least this PR won't cause another performance regression right?

From an overview, doesn't look like there will be a regression on main but we need to regression test here because in this PR we have some changes to the Freeze wrapper. 

> this is already existing bug on main right?

That's correct and this is what's addressed in this [PR](https://github.com/Expensify/App/pull/36420)

**Comment by perunt** [comment]
Ahh, now I understand—I was concerned my PR might be causing some slowness. Great work on your part! The page loading appears much faster now. I'm confident users will definitely notice the improvemen

**Comment by perunt** [comment]
> @hurali97 at least this PR won't cause another performance regression right?

Indeed, this PR should not lead to any further performance regression. At least, I haven't observed any such issues so far.

**Comment by perunt** [comment]
@situchan, can you take a look at this?

**Comment by perunt** [comment]
@situchan, can you please take a look at this?

**Comment by situchan** [comment]
yes reviewing today

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by situchan** [comment]
@perunt can you please add platform screenshots?

**Comment by situchan** [comment]
And there's conflict

**Comment by perunt** [comment]
> @perunt can you please add platform screenshots?

it was a web issue, i have screen records for it in `MacOS: Chrome / Safari` section

**Comment by situchan** [comment]
still testing to find possible regressions.
Please merge main. I will try to complete review today

File: src/libs/Navigation/FreezeWrapper.tsx (hunk 2 above)
Comment (adhorodyski) [reply src/libs/Navigation/FreezeWrapper.tsx]: FYI I also gave it a spin and wasn't able to experience anything weird related to animations.

**Comment by situchan** [comment]
This also happens on main so out of scope. But is this bug already reported or being fixed elsewhere?

After switching to wrench tab and back, not found page shows

https://github.com/Expensify/App/assets/108292595/15687eb9-7a01-4ac8-8db7-6e50301f9200

File: src/libs/Navigation/FreezeWrapper.tsx [hunk 3]
```diff
@@ -28,8 +27,9 @@ function FreezeWrapper({keepVisible = false, children}: FreezeWrapperProps) {
             // if the screen is more than 1 screen away from the current screen, freeze it,
             // we don't want to freeze the screen if it's the previous screen because the freeze placeholder
             // would be visible at the beginning of the back animation then
-            if ((navigation.getState()?.index ?? 0) - (screenIndexRef.current ?? 0) > 1) {
-                InteractionManager.runAfterInteractions(() => setIsScreenBlurred(true));
+
+            if ((navigation.getState()?.index ?? 0) - (screenIndexRef.current ?? 0) >= 1) {
```
Comment (situchan) [inline src/libs/Navigation/FreezeWrapper.tsx:31]: I see that you changed to `>= 1` from `> 1`. Isn't this opposed to this comment?
```
    // we don't want to freeze the screen if it's the previous screen because the freeze placeholder
    // would be visible at the beginning of the back animation then
```

File: src/libs/Navigation/FreezeWrapper.tsx [hunk 4]
```diff
@@ -28,8 +27,9 @@ function FreezeWrapper({keepVisible = false, children}: FreezeWrapperProps) {
             // if the screen is more than 1 screen away from the current screen, freeze it,
             // we don't want to freeze the screen if it's the previous screen because the freeze placeholder
             // would be visible at the beginning of the back animation then
-            if ((navigation.getState()?.index ?? 0) - (screenIndexRef.current ?? 0) > 1) {
-                InteractionManager.runAfterInteractions(() => setIsScreenBlurred(true));
```
Comment (situchan) [inline src/libs/Navigation/FreezeWrapper.tsx:32]: > InteractionManager.runAfterInteractions

@getusha can you please check if this removal is fine?
I see you added this workaround during RNW upgrade [here](https://github.com/Expensify/App/commit/0b4ee5c396a14b07a464b70f47e696f5095e2c58). To fix what bug?

**Comment by situchan** [comment]
It doesn't make sense to link https://github.com/Expensify/App/issues/35602 in `Fixed Issues` as this PR isn't supposed to that.

File: src/libs/Navigation/FreezeWrapper.tsx (hunk 3 above)
Comment (perunt) [reply src/libs/Navigation/FreezeWrapper.tsx:31]: I see that we no longer use the placeholder component. I guess this comment was not changed after the placeholder was removed. I will do it now

**Comment by perunt** [comment]
> It doesn't make sense to link https://github.com/Expensify/App/issues/35602 in Fixed Issues as this PR isn't supposed to that.

switched to the correct one https://github.com/Expensify/App/pull/36420#issuecomment-1946291081

@situchan, can I somehow speed up merging this PR?

**Comment by situchan** [comment]
> @situchan, can I somehow speed up merging this PR?

I am just waiting for https://github.com/Expensify/App/pull/36927#discussion_r1517749523 from @getusha to confirm that it's safe.
Otherwise looks good.

File: src/libs/Navigation/FreezeWrapper.tsx (hunk 3 above)
Comment (situchan) [reply src/libs/Navigation/FreezeWrapper.tsx:27]: @perunt this actually caused regression.
As this affects all platforms, please test and add remaining platforms, not only web.


this branch:

https://github.com/Expensify/App/assets/108292595/4cf409eb-f134-4c5a-9cd6-79c0bcdf95fd


main:

https://github.com/Expensify/App/assets/108292595/9b6ed6de-ee9d-4af6-81e9-56e649259b09

**Review by hayata-suenaga — APPROVED** [review]
code looks good to me

are there blocking issues or conversations to be resolved before we merge this PR?

**Comment by perunt** [comment]
I don't know about such issues, so it's clean from my side

**Comment by situchan** [comment]
> code looks good to me
> 
> are there blocking issues or conversations to be resolved before we merge this PR?

https://github.com/Expensify/App/pull/36927#discussion_r1520078655

**Comment by hayata-suenaga** [comment]
@getusha could you answer the question from @situchan ?

**Comment by hayata-suenaga** [comment]
@perunt please check the regression described in [this comment](https://github.com/Expensify/App/pull/36927#discussion_r1520078655).

**Review by hayata-suenaga — CHANGES_REQUESTED** [review]
@perunt please check the regression described in [this comment](https://github.com/Expensify/App/pull/36927#discussion_r1520078655).

File: src/libs/Navigation/FreezeWrapper.tsx (hunk 4 above)
Comment (hayata-suenaga) [reply src/libs/Navigation/FreezeWrapper.tsx:32]: answered [here](https://expensify.slack.com/archives/C05LX9D6E07/p1710172332635789?thread_ts=1710170982.916159&cid=C05LX9D6E07)

File: src/libs/Navigation/FreezeWrapper.tsx (hunk 3 above)
Comment (perunt) [reply src/libs/Navigation/FreezeWrapper.tsx:27]: @situchan, thanks for pointing it out. 
I retested it, and it looks like a freshly added nullish coalescing operator can let me remove this check, so I assume we can't receive navigation immediately in some cases. I'm adding the rest of the recordings.

**Comment by perunt** [comment]
@hayata-suenaga I'm resolving this, thanks!

**Comment by hayata-suenaga** [comment]
@situchan could you retest when you have time? 🙇

**Comment by situchan** [comment]
testing

**Review by situchan — APPROVED** [review]

**Comment by situchan** [comment]
The code is now simplified and affects only web platform.
Tests well but still be on the lookout for any regressions we might have missed.

**Review by hayata-suenaga — APPROVED** [review]
thank you so much for your work @perunt 😄
